daemon/network.go

```diff
@@ -96,15 +96,7 @@ func (daemon *Daemon) GetNetworksByID(partialID string) []libnetwork.Network {
 
 // getAllNetworks returns a list containing all networks
 func (daemon *Daemon) getAllNetworks() []libnetwork.Network {
-	c := daemon.netController
-	list := []libnetwork.Network{}
```

Comment:
Thanks @allencloud You are right about the return list now can be a nil.

But in regards to the HTTP APIs, given they have to return a list of `NetworkResource` objects, the router function `getNetworksList()` https://github.com/docker/docker/blob/master/api/server/router/network/network_routes.go#L44 
prepares another list and handle the nil returned by `GetNetworks()` (read `getAllNetworks()`) properly as part of the range loop: https://github.com/docker/docker/blob/master/api/server/router/network/network_routes.go#L53